src/Datadog.Trace.ClrProfiler.Managed/Integrations/ElasticsearchNet5Integration.cs

```diff
@@ -110,6 +111,7 @@ public static object CallElasticsearchAsync<TResponse>(object pipeline, object r
                        .WithConcreteType(pipeline.GetType())
                        .WithMethodGenerics(genericArgument)
                        .WithParameters(requestData, cancellationToken)
+                       .WithNamespaceAndNameFilters(ClrNames.Task, "Elasticsearch.Net.RequestData", ClrNames.CancellationToken)
```

Comment:
Will `ClrNames.Task` work even though the method's return type is `Task<T>`, not `Task`? Couldn't we end up calling the wrong overload? Same for all the other `async` methods.

Reply:
We could end up calling the wrong overload generally... But I think not in this case, because it's the return type.
This does require some thought, but would you agree this is much safer than before?
Especially because this is one layer among others.
The checks look at:
name matches -> explicit return type if specified number of parameters -> namespace name filters -> all non-null parameters are assignable -> generic counts match if applicable -> generic constraints are satisfied if applicable

Reply:
>  would you agree this is much safer than before?

Definitely better. I just wanted to make sure that we were aware of this (the return type not being an exact match) and it wasn't a oversight.